packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -1197,6 +1320,68 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
             // This highlights the left side if the row is loading.
             this.setState({ selectedRow: id });
         }
+
+        if (this.state.ctxMenu) {
+            this.setState({ selectedRows: [id] });
+            signalManager().fireRowSelectionsChanged({
+                traceId: this.props.traceId,
+                outputDescriptor: this.props.outputDescriptor,
+                rowIds: [id]
+            });
+        }
+    };
+
+    public onMultipleRowClick = (id: number, isShiftClicked?: boolean): void => {
+        // Hoooked to a context menu - if no context menu provided, multi-selection will route to onRowClick
```

Comment:
**Note**  multi selection is hooked to if a context menu was registered - without a context menu, multi-selection is not needed IMO (but If there is a use case that anyone thinks of - i can remove this check)